### ISSUE TITLE
Enhancement: Extend HtmlChangeListener to also include the rejected content/value

### ISSUE THREAD

**Issue by ghost** [OWASP/java-html-sanitizer#243]
Something like the following would be really nice:

```
public interface HtmlChangeListener<T> {

  /** Called when a tag is discarded from the input. */
  public void discardedTag(@Nullable T context, String elementName, String rejectedContent);

  /**
   * Called when attribute is discarded
   * from the input but the containing tag is not.
   */
  public void discardedAttribute(
      @Nullable T context, String tagName, String attributeName, String rejectedValue);
}
```

**Comment by lread** [comment]
I'd like something like this too.

It is great to have the listener support, but without more detail, it is difficult to easily pinpoint exactly why something was discarded, especially for larger content.

If there is a reason why including rejected values might not be a good idea, a reasonable alternative might be to share the row/col offset (from original content) of where the issue occurred with the listener.

**Comment by lread** [comment]
As I'm using java-html-sanitizer more it is occurring to me, at least in my current usage, that knowing the discarded value/content is only helpful when the item being rejected is rejected due to its content/value.

If an element or attribute is rejected outright (not due to its value/content), knowing its value/content does not add information. 

In my current usage, only attributes are rejected due to their value.

**Comment by lread** [comment]
Thanks, @jmanico!

**Comment by jmanico** [comment]
Please let me know if the new release is working for you. It's a major major change and I'm a bit nervous.

**Comment by lread** [comment]
Yes, it is a whopper! Thanks for all the hard work! When I give it a try, I'll post back here.

**Comment by jmanico** [comment]
"What issues and bugs did we fix?"

**_<marquee><blink>ALL OF THEM</blink></marquee>_**
